ajbleakley/register-app
# Feature - Install Doctrine

### What is the (feature|problem)?

Relates to: [AS AN application I WANT TO use an ORM database tool SO THAT I can easily query and manipulate data from a database](https://github.com/ajbleakley/register-app/issues/15)

### What is the solution?

[Proposed solution](https://github.com/ajbleakley/register-app/issues/15) was to (a) install [Doctrine ORM](https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/index.html), (b) configure a [PDO_SQLITE](https://www.php.net/manual/en/ref.pdo-sqlite.php) driver and (c) define a starting point for a user entity.

### What areas of the app does it impact?

Allows the application to persist data to a database while remaining database agnostic. 

### How to test?
1. Rebuild your application images and spin up your containers:
```
docker compose up --build -d
```
2. To verify Doctrine CLI is installed and to view a list of available commands:
```
docker exec -it register-app-php-fpm-1 composer doctrine
```
3.  To confirm (starting point for) mapped entity in place for user:
```
docker exec -it register-app-php-fpm-1 composer doctrine orm:info
```
